alphagov/gsp-harbor-docker-image-resource
# Try and Catch problems with notary setup

## What

Try and Catch problems with notary setup clean up after failure.

We have been beaten before, by harbor going down before notary manages
to finish up the setup. This causes issues later as the setup didn't add
the delegation key to the server, meaning the target key generated by
the init command is required.

We choose to dispose the target keys, therefore we need to clean up the
current state in preparation for a re-try and not block the users.

I don't think the actions need to be separated out. I would of hoped
only that block is capable of destroying the notary setup and only that
block (if goes wrong) can do any harm.

## How to review

- Code review
- Sanity check
- Run the resource locally
- Run the resource locally - but try to fail at some point (misspell `dockernotary`)
- Run the resource locally - Make sure your image is being pushed
  - I haven't tested that part
